safe/common/parameters/default_select_parameter_widget.py

```diff
@@ -151,3 +152,29 @@ def toggle_custom_value(self):
             self.custom_value.setDisabled(False)
         else:
             self.custom_value.setDisabled(True)
+
+    def toggle_input(self):
+        """Change behaviour of radio button based on input."""
+        # If input is a field, then disable all radio button.
+        if self.input.currentText() != 'No Field':
```

Comment:
Be careful 'No Field' will be translated in the UI, so you won't get this string. You should translate it here too. But I would prefer if you have 'data' (http://doc.qt.io/qt-4.8/qcombobox.html#itemData) in the combobox. I don't like string comparaison from the UI (we got some issues in the past between french, bahasa, ...).